modules/redis/redis.h

```diff
@@ -34,6 +34,9 @@ void redis_dd_set_auth(LogDriver *d, const gchar *auth);
 void redis_dd_set_command(LogDriver *d, const gchar *command,
                           LogTemplate *key,
                           LogTemplate *param1, LogTemplate *param2);
+void redis_dd_set_command_ref(LogDriver *d, const gchar *command,
```

Comment:
Does it really make sense to create a new function which calls the original, and now the original is not used only by the new function ? It would if it were some kind of API, but it is not.

The whole think just gets too complicated, what is a downside of assuming that the set function grabs ownership as well ?

Reply:
Agreed, I will fix it as Bazsi suggested :)

Reply:
I updated the PR with renaming the `set` function and unrefing the templates.
Could I just remove the `log_template_ref` calls and not even call `unref` later? It could be easier to read.

Reply:
I see now, if i remove the `ref` call, `g_assert(s->ref_cnt > 0);` will not be checked, so I guess it is needed after all.

Reply:
Thanks! I think I finally have the desired fix :D

Reply:
Thanks.